terraso_backend/apps/graphql/schema/commons.py

```diff
@@ -49,6 +51,15 @@ class Meta:
 
     @classmethod
     def mutate(cls, root, info, input):
+        user = info.context.user
+
+        if not user or not user.is_authenticated:
+            message = {
+                "message": "You must be authenticated to perform this operation",
+                "code": "unauthorized",
+            }
+            return cls(errors=[{"message": json.dumps([message])}])
+
```

Comment:
suggestion: Would this be better implemented as [middleware](https://docs.graphene-python.org/en/latest/execution/middleware/) ([graphene-django](https://docs.graphene-python.org/projects/django/en/latest/settings/#middleware))? 

I was having some problems with this function yesterday (#394 , still have to fill out the details). I think the functionality it tries to perform would be better implemented with middleware. The class hierarchy it relies on in `graphene-django` is not documented and it's difficult for me to figure out why it's not working as expected.

This is kind of a big ask for this PR, so I'd be OK with adding a comment somewhere someone adding a mutation will see it (i.e. `schemas/__init__.py`, above the `Mutations` class) explaining that all Mutations MUST extend this base class. I say this because I was considering bypassing this base class yesterday in some of the LandPKS code I was writing, and that would not be good.

Reply:
@david-code I would like to get more context about this. This right now is just related to mutations, so this validation is inside the Graph QL schema definition, right now it is working for all mutations, but we don't know if in the future we want some types of mutations that could be accessed anonymously, still it seems that a validation at the mutation level is something that could be done. So if the issue is being able to identify from where the error generates I think that is more related to our logs, because even if we move this validation outside of the mutations, you can defenitely add validations inside mutations. 

I think the comment where we have the schame defined saying that all mutations should use it is a good thing, I am not sure if there is a way to enforce it, I was thinking about something like that, like creating custom rules

Reply:
I see, sorry, I wasn't very clear. When I said "I'm not sure why it isn't working as expected", I meant that I had problems yesterday working with this function (without your changes). It doesn't seem to correctly return the error message I expect. The rest of the comment was just noting that we are using a feature that isn't documented by `graphene-django`, namely that `mutate_and_get_payload` is called by the `mutate` method in the parent class here. I found this confusing yesterday when I was working with the code.

Instead of overriding this method to try and perform authentication checks, I was thinking of basically doing the same thing in some middleware. This would have the benefit that it would automatically apply to any mutation created in the future, without needing to extend this base class. 

However, I realize that is a bit of a complicated process, and after all I'm not sure if you agree with me that the solution is optimal. In that case, I think just adding a comment to the `__init__` file is fine; if you want to add a new mutation, you need to read that file anyways. This would just avoid someone writing a Mutation that is accidentally exposed to unauthenticated users by not extending the base class.